platform/default/src/mbgl/storage/offline_database.cpp

```diff
@@ -76,30 +76,36 @@ void OfflineDatabase::changePath(const std::string& path_) {
     initialize();
 }
 
-void OfflineDatabase::cleanup() {
+void OfflineDatabase::cleanup(bool deleteDBFile) {
```

Comment:
I think it should return whether cleanup was successful. As the exception for deletion is caught here, ::initialize can go into infinite recursion loop.

Reply:
Fixed!